### PR TITLE
feat(rules): support CIDR ranges in admin-ips

### PR DESCRIPTION
**Pull request by selfuryon**
Now admin-ips entries may now be individual IP addresses or CIDR ranges; a plain IP is treated as a CIDR covering that address alone. Invalid entries are now rejected at startup instead of silently never matching.

### CODE REVIEW

File: rules/standard/service.go [hunk 1]
```diff
@@ -41,15 +42,20 @@ func New(ctx context.Context, params ...Parameter) (*Service, error) {
 		log = log.Level(parameters.logLevel)
 	}
 
+	adminIPNets, err := parseAdminIPs(parameters.adminIPs)
+	if err != nil {
+		return nil, errors.Wrap(err, "problem with admin IPs")
+	}
+
```
Comment (AntiD2ta) [inline rules/standard/service.go]: There was probably a good reason why Jim never checked for the `adminIP` correctness here at `Service` construction time. I think that reason is avoiding not letting Dirk's daemon initialize due to a non essential config issue. 

It is better to find out that a voluntary exit's signing doesn't work during runtime than have Dirk's instances down and able to start due to a wrong config value after an update/restart.

That being said, I find this new IP checks very positive, I just don't think is wise for them to block the daemon. How do you feel about a combo of:

- Logging wrong entries
- Skipping wrong entries

All of that here, but without returning an error.

Comment (AntiD2ta) [reply rules/standard/service.go]: Prior to the CIDRs, an IP address could be wrongly configured (regarding correctness), but since it would never match a correct IP address, this wasn't a big issue. We were not skipping or logging it before, therefore for operators it was not straightforward to diagnose the problem.

File: rules/standard/adminips.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+// parseAdminIPs turns the configured admin IP entries in to IP networks that can be
+// checked against an incoming request IP address.  Each entry may be a single IP
+// address, e.g. "1.2.3.4", or a CIDR range, e.g. "10.0.0.0/8"; a single IP address is
+// treated as a CIDR range covering that address alone.
+func parseAdminIPs(entries []string) ([]*net.IPNet, error) {
+	ipNets := make([]*net.IPNet, 0, len(entries))
+	for _, entry := range entries {
+		if strings.Contains(entry, "/") {
+			_, ipNet, err := net.ParseCIDR(entry)
```
Comment (AntiD2ta) [inline rules/standard/adminips.go]: Flagged by Claude:

> `net.ParseCIDR` discards non-zero host bits rather than rejecting them. An operator who configures "10.1.2.3/24" intending to allow only host 10.1.2.3 gets network 10.1.2.0/24 back with no error — verified: `ipNet.Contains(net.ParseIP("10.1.2.99")) == true`.
> 
> `adminIPAllowed` then approves voluntary-exit requests from any of the 254 hosts in that /24, silently expanding the trusted set for validator exits well beyond what the operator intended, with no error or log to flag the mismatch. Under the pre-PR string-equality code this same entry matched nothing, so this over-permissive behavior is new attack surface introduced by the diff.

Is that what we want here? Assuming is correct. I suggest adding more test cases to cover such edge scenarios.

**Review by AntiD2ta — CHANGES_REQUESTED** [review]
Happy to see this progressing.

Apart from the other comments, could we add test cases (on both `rules_tests.go` and `sign_test.go`) for a (non-CIDR) IPv6 address?

File: rules/standard/service.go (hunk 1 above)
Comment (selfuryon) [reply rules/standard/service.go]: Yeah, make sense, let me make it like before

File: rules/standard/adminips.go (hunk 2 above)
Comment (selfuryon) [reply rules/standard/adminips.go]: Hm, it's interesting. Formally `10.1.2.3/24` == `10.1.2.0/24` and it isn't error to write like that. But I agree it looks like an error in mask in such cases, it's rare that people write like that so we can be more strict than spec here. Let me do that

File: rules/standard/service.go (hunk 1 above)
Comment (selfuryon) [reply rules/standard/service.go]: fixed

File: rules/standard/adminips.go (hunk 2 above)
Comment (selfuryon) [reply rules/standard/adminips.go]: Added

**Review by Bez625 — APPROVED** [review]

**Review by AntiD2ta — APPROVED** [review]
